src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -152,7 +152,7 @@ public SegmentInfos readLastCommittedSegmentsInfo() throws IOException {
      */
     private static SegmentInfos readSegmentsInfo(IndexCommit commit, Directory directory) throws IOException {
         try {
```

Comment:
Maybe remove `Directory directory` arg here?  Or assert that `directory == commit.getDirectory()`?